exchange_calendars/exchange_calendar_xkrx.py

```diff
@@ -569,18 +64,195 @@ class XKRXExchangeCalendar(PrecomputedExchangeCalendar):
+        (pd.Timestamp("1987-03-01"), time(11, 40)),
+        (pd.Timestamp("1995-01-01"), time(11, 30)),
+        (pd.Timestamp("1998-12-07"), time(12, 0)),
+        (pd.Timestamp("2000-05-22"), None),
+    )
+    break_end_times = (
+        (None, time(13, 30)),
+        (pd.Timestamp("1986-04-01"), time(13, 20)),
+        (pd.Timestamp("1995-01-01"), time(13, 0)),
+        (pd.Timestamp("2000-05-22"), None),
+    )
+    close_times = (
+        (None, time(15, 30)),
+        (pd.Timestamp("1986-04-01"), time(15, 20)),
+        (pd.Timestamp("1995-01-01"), time(15, 0)),
+        (pd.Timestamp("2016-08-01"), time(15, 30)),
+    )
+
+    # Saterday became holiday since 1998-12-07
+    # https://www.hankyung.com/finance/article/1998080301961
+
+    weekmask = "1111100"
 
     @property
-    def name(self):
-        # Korea Exchange
-        return "XKRX"
+    def special_weekmasks(self):
+        return [
+            (None, pd.Timestamp("1998-12-07") - pd.Timedelta(1, unit="D"), "1111110"),
+        ]
+
+    # KRX regular and adhoc holidays
+
+    @property
+    def regular_holidays(self):
+        return HolidayCalendar(krx_regular_holiday_rules)
+
+    @property
+    def adhoc_holidays(self):
+        return precomputed_krx_holidays.tolist()
+
+    # The first business day of each year:
+    #  opening schedule is delayed by an hour.
+
+    @property
+    def special_offsets(self):
+        return [
+            (
+                pd.Timedelta(1, unit="h"),
+                None,
+                None,
+                None,
+                HolidayCalendar(
+                    [
+                        Holiday(
+                            "First Business Day of Year",
+                            month=1,
+                            day=2,
+                            observance=next_monday,
+                        )
+                    ]
+                ),  # avoiding 01/01 since it's New Year's Day
+            ),
+        ]
+
+    # Every year's CSAT day, all schedules are delayed by:
+    #  before 1998-11-18: 30 minutes
+    #  after  1998-11-18: 1 hour
+
+    @property
+    def special_offsets_adhoc(self):
+        return [
+            (
+                pd.Timedelta(30, unit="m"),
+                pd.Timedelta(30, unit="m"),
+                pd.Timedelta(30, unit="m"),
+                pd.Timedelta(30, unit="m"),
+                precomputed_csat_days[
+                    precomputed_csat_days.slice_indexer("1993-08-20", "1998-11-17")
+                ],
+            ),
+            (
+                pd.Timedelta(1, unit="h"),
+                pd.Timedelta(1, unit="h"),
+                pd.Timedelta(1, unit="h"),
+                pd.Timedelta(1, unit="h"),
+                precomputed_csat_days[
+                    precomputed_csat_days.slice_indexer("1998-11-18", None)
+                ],
+            ),
+        ]
+
+
+class PrecomputedXKRXExchangeCalendar(PrecomputedExchangeCalendar):
```

Comment:
Any reason to keep this one around?

Reply:
For those who may want the original behavior for any reason.